public/static/app/perf-bb/PerfBBOutliersCtrl.js

```diff
@@ -395,4 +190,373 @@ mciModule.controller('PerfBBOutliersCtrl', function (
+        })
+        .then(success, error)
+        .finally(() => this.lock.unlock())
+        .catch($log.debug);
+    };
+
+    unmuteOutliers(success, error) {
+      const items = this.state.selection;
+      this.lock.lock();
+      this.confirmUnmuteAction(items)
+        .then(() => {
+          const promises = _.map(items, function (item) {
+            const mute = MuteHandler.getMuteIdentifier(item);
+            if (item.muted) {
+              return MuteDataService.unMute(mute);
+            } else {
+              return mute;
+            }
+          });
+          return $q.all(promises)
+        })
+        .then(success, error)
+        .finally(() => this.lock.unlock())
+        .catch($log.debug);
+    };
+
+    static getMuteIdentifier(mute) {
+      return _.pick(mute, "project", "revision", "task", "test", "thread_level", "variant", "create_time", "end", "last_updated_at", "order", "task_id", "version_id");
+    }
+
+  }
+
+  return MuteHandler;
+}).factory('OutlierState', function(FORMAT, MDBQueryAdaptor, EvgUtil, EvgUiGridUtil, uiGridConstants, OutliersDataService, MuteDataService, $q) {
+  class OutlierState {
+    // Create a new state instance to encapsulate filtering, sorting and generating queries.
+    //
+    // :param str project: The project name, e.g. 'sys-perf'.
+    // :param object model: The view model instance.
+    // :param object scope: The $scope instance.
+    // :param dict mandatory: The mandatory filter fields / values.
+    // :param list transient: The transient filter fields. These fields are never persisted to the settings.
+    // :param list(dict) sorting: The default fields to sort on.
+    // :param object settings: A reference to the persistent settings instance.
+    // :param int units: The look back value. Defaults to 2.
+    // :param str units: The look back units. Defaults to 'weeks'.
+    constructor(project, model, mandatory, transient, sorting, settings, limit, value = 2, units = 'weeks') {
+      this.project = project;
+      this.vm = model;
+      this.mandatory = mandatory;
+      this.transient = transient;
+      this._sorting = sorting;
+      this.settings = settings;
+      this.limit = limit;
+
+      this.lookBack = moment().subtract(value, units);
+      this.getCol = null;
+      this.api = null;
+    }
+
+    // Handle a sort change event.
+    // :param object grid: Not used.
+    // :param object cols: The column definitions.
+    onSortChanged(grid, cols) {
+      this._sorting = _.map(cols, function (col) {
+        return {
+          field: col.field,
+          direction: col.sort.direction
+        };
+      });
+    }
+
+    // Handle a rowsRendered event. This is only expected to be called once.
+    // :param object api: The grid api reference.
+    // :param object cols: The column definitions.
+    onRowsRendered(api) {
+      this.api = api;
+      this.getCol = EvgUiGridUtil.getColAccessor(api);
+
+      //const setInitialGridFiltering = () => {
```

Comment:
[nit] Can the commented out lines in this function be deleted?

Reply:
I will update to have a proper comment that explains the intent. (ditto for sorting).